src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ParameterNamesShouldMatchBaseDeclaration.cs

```diff
@@ -4,15 +4,20 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Analyzer.Utilities;
+using System;
+using System.Linq;
+using System.Collections.Generic;
 
 namespace Microsoft.ApiDesignGuidelines.Analyzers
 {
     /// <summary>
     /// CA1725: Parameter names should match base declaration
     /// </summary>
-    public abstract class ParameterNamesShouldMatchBaseDeclarationAnalyzer : DiagnosticAnalyzer
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class ParameterNamesShouldMatchBaseDeclarationAnalyzer : DiagnosticAnalyzer
     {
         internal const string RuleId = "CA1725";
+        internal const string NewNamePropertyName = "NewName";
 
         private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.ParameterNamesShouldMatchBaseDeclarationTitle), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
 
```

Comment:
This is not your fault, but the message resource string `ParameterNamesShouldMatchBaseDeclarationMessage` is not correct - it should be parameterized with the actual method name. As we are trying to retain as much compat with FXCop as possible, you should change it to:

```
In member {0}, change parameter name {1} to {2} in order to match the identifier as it has been declared in {3}.
```
where {0} = violating member, {1} = violating parameter name, {2} = base parameter name, {3} = base member